doc/mgr/prometheus.rst

```diff
@@ -92,25 +92,25 @@ RBD IO statistics
 
 The module can optionally collect RBD per-image IO statistics by enabling
 dynamic OSD performance counters. The statistics are gathered for all images
-in the pools that are specified in the ``mgr/prometheus/rbd_stats_pools``
+in the pools that are specified in the ``mgr/prometheus/<mgr-id>/rbd_stats_pools``
 configuration parameter. The parameter is a comma or space separated list
 of ``pool[/namespace]`` entries. If the namespace is not specified the
 statistics are collected for all namespaces in the pool.
 
 Example to activate the RBD-enabled pools ``pool1``, ``pool2`` and ``poolN``::
 
-  ceph config set mgr mgr/prometheus/rbd_stats_pools "pool1,pool2,poolN"
+  ceph config set mgr mgr/prometheus/<mgr-id>/rbd_stats_pools "pool1,pool2,poolN"
 
 The module makes the list of all available images scanning the specified
 pools and namespaces and refreshes it periodically. The period is
-configurable via the ``mgr/prometheus/rbd_stats_pools_refresh_interval``
+configurable via the ``mgr/prometheus/<mgr-id>/rbd_stats_pools_refresh_interval``
 parameter (in sec) and is 300 sec (5 minutes) by default. The module will
 force refresh earlier if it detects statistics from a previously unknown
 RBD image.
 
 Example to turn up the sync interval to 10 minutes::
 
-  ceph config set mgr mgr/prometheus/rbd_stats_pools_refresh_interval 600
+  ceph config set mgr mgr/prometheus/<mgr-id>/rbd_stats_pools_refresh_interval 600
```

Comment:
@trociny hi Mykola, i am wondering why we choose to use localized setting for these options? for instance, back in ad174de0808981d733508b6dc2d528eb6cd96519 , `rbd_stats_pools_refresh_interval` was introduced. i think the per-mgr options are used for per-mgr settings like server address and port, but the settings like `rbd_stats_pools_refresh_interval` should be global settings. shall we instead use `self.get_module_option()` instead ?

Reply:
`get_localized_module_option` will use the global setting if the localized is not used. So I did not see anything wrong with using `get_localized_module_option` for options which are usually expected to be set on global level, still leaving a chance to set it per mgr (if it is needed for some strange reason).

I don't have a strong opinion about this and can change this or feel free to change this yourself. It would be good if we had some convention about this used for all modules.

It is definitely wrong to update the documentation though. The old version is correct: although it is possible to set it per mgr, it is not what users will usually want.

Reply:
Sorry about the pr without careful thoughts and enough contexts. So should I close the pr now, or narrow the scope of modification, since global settings work for each mgr.

Reply:
@trociny thanks for your insights. neither do i feel strong regarding using localized settings for storing options like "scrape_interval". but from user's point of view, it'd be confusing. as i always consider multiple instances of mgr is a way of failure tolerance, but not an approach to offer this kind of flexibility, on the contrary it could lead to different behaviors when switching from one mgr instance to another. but that's not in the scope of this change.

>  It would be good if we had some convention about this used for all modules.

indeed. will find some time to improve the document.

and i agree with you the existing command line samples like

```
ceph config set mgr mgr/prometheus/scrape_interval 20
```

are correct. but does it make sense to note in the document that per-mgr settings of ``mgr/prometheus/<mgr-id>/server_addr`` and ``mgr/prometheus/<mgr-id>/server_port`` are offered to allow user to specify different scrape targets for its file-based service discovery if we don't have an HA proxy in front of mgr instances?

@Rethan sorry for the bikeshedding. probably latter -- to s/config-key/config/.

Reply:
> neither do i feel strong regarding using localized settings for storing options like "scrape_interval". but from user's point of view, it'd be confusing.

The same thing is our "common" config options -- you can set it globally for all daemons or just per one daemon in its section, and for some variables it does not make much sense, still we allow this.

> does it make sense to note in the document that per-mgr settings of mgr/prometheus/<mgr-id>/server_addr and mgr/prometheus/<mgr-id>/server_port are offered to allow user to specify different scrape targets for its file-based service discovery if we don't have an HA proxy in front of mgr instances?

I am not sure it is worth mentioning it in the doc, especially if you have plans to revisit this behavior. It just might make thing more confusing in the future for users if the behavior will change.

May be it should be a more generic statement, for all modules, not only prometheus, that for some (many?) module options it is supported to specify them per-mgr, though usually it is not expected to be useful and one need to check if it is applied for the specific option that one wants to change.

Reply:
> 
> @Rethan sorry for the bikeshedding. probably latter -- to s/config-key/config/.
It's ok. Hope to see more consistent configuration ways (and docs) in the future.

Reply:
> May be it should be a more generic statement, for all modules, not only prometheus, that for some (many?) module options it is supported to specify them per-mgr, though usually it is not expected to be useful and one need to check if it is applied for the specific option that one wants to change.

agreed. adding it to my backlog.